tests/unittests/algo/test_base.py

```diff
@@ -108,3 +111,39 @@ def test_is_done_max_trials(monkeypatch, dumbalgo):
 
     dumbalgo.max_trials = 4
     assert algo.is_done
+
+
+@pytest.mark.parametrize("pass_to_super", [True, False])
+def test_arg_names(pass_to_super: bool):
+    """Test that the `_arg_names` can be determined programmatically when the args aren't passed to
+    `super().__init__(space, **kwargs)`.
+
+    Also checks that the auto-generated configuration dict acts the same way.
+    """
+    # NOTE: If there is a **kwargs argument in __init__ signature, and it is not passed to the
+    # super().__init__(self, **kwargs), then what should the _arg_names be? Should we just raise a
+    # warning in that case? Or just assume that the algo will create its own configuration dict?
+    # (could also raise a warning only inside the `configuration` property, in the case where we
+    # can't fetch the value of the kwargs argument)
```

Comment:
Algo configuration is expected to be compliant with algorithm's constructor signature, so that we can build an algorithm with `AlgoClass(space, **configuration)`. I think we could safely ignore any of kwargs and rely on the signature. 

Also, setting the attributes by passing them to super() is always very confusing to new contributiors. I would be in favor of deprecating this and force algos to set their attributes themselves using `self.attr = attr`.

Reply:
Ok, so you're ok with these changes then? (I'll remove the comment)